src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/ProjectDependenciesSubTreeProvider.cs

```diff
@@ -80,16 +89,68 @@ protected override IDependencyNode CreateRootNode()
                                                                 IImmutableDictionary<string, string> properties = null,
                                                                 bool resolved = true)
         {
+
+            var projectPath = itemSpec;
+            if (!Path.IsPathRooted(projectPath))
+            {
+                var currentProjectFolder = Path.GetDirectoryName(UnconfiguredProject.FullPath);
+                projectPath = Path.GetFullPath(Path.Combine(currentProjectFolder, projectPath));
+            }
+
             var id = new DependencyNodeId(ProviderType,
                                           itemSpec,
-                                          itemType ?? ResolvedProjectReference.PrimaryDataSourceItemType);
+                                          itemType ?? ResolvedProjectReference.PrimaryDataSourceItemType,
+                                          uniqueToken: projectPath);
             return new ProjectDependencyNode(id,
                                              flags: ProjectSubTreeNodeFlags,
                                              priority:priority,
                                              properties: properties,
                                              resolved: resolved);
         }
 
+        public override IDependencyNode GetDependencyNode(DependencyNodeId nodeId)
+        {
+            if (nodeId == null)
+            {
+                return null;
+            }
+
+            // normalize id to have regular paths (graph provider replaces \ with /.
+            nodeId = nodeId.ToNormalizedId();
+            var node = RootNode.FindNode(nodeId, recursive: true);
+            if (node == null)
+            {
+                return null;
+            }
+
+            if (string.IsNullOrEmpty(nodeId.UniqueToken))
+            {
+                return node;
+            }
+
+            var projectPath = nodeId.UniqueToken;
+            var projectContext = ProjectContextProvider.GetProjectContext(projectPath);
+            if (projectContext == null)
+            {
+                return node;
+            }
+
+            foreach (var subTreeProvider in projectContext.GetProviders())
```

Comment:
Add a change to refresh project dependency chnages

---
In reply to: [86005324](https://github.com/dotnet/roslyn-project-system/pull/675#discussion_r86005324) [](ancestors = 86005324)